wire/msgtx.go

```diff
@@ -586,8 +587,9 @@ func (msg *MsgTx) BtcDecode(r io.Reader, pver uint32, enc MessageEncoding) error
 			// item itself.
 			txin.Witness = make([][]byte, witCount)
 			for j := uint64(0); j < witCount; j++ {
-				txin.Witness[j], err = readScript(r, pver,
-					maxWitnessItemSize, "script witness item")
+				txin.Witness[j], err = readScript(
```

Comment:
I am wondering why we do not enforce the different size limits for the different segwit version here ? For v_0 we stay with 11kb and for 4MB ?

Reply:
So that check is deeper in the consensus validation logic, this is a wire level DDoS defense limit/value.